krizten/Memoire-WebAPI
# #164977335 Add Forgot Password Endpoint for Memoire API

### What does this PR do?
- This allow users to reset their passwords they have forgotten.
#### Description of Task to be completed?
- Get the following endpoints working
```POST /api/v1/auth/forgot-password```
```POST /api/v1/auth/reset-password?key=<resetToken>```
#### How should this be manually tested?
* Clone this repository and checkout to this branch (ft/164977335/forgot-password-endpoint).
* Install the Postgres database by following the instruction on the README.
* Install the depedencies by running ```npm install```
* Run the application in development by running ```npm run start:dev```
* Using a REST Client such as Postman, create a user account using the ```POST api/v1/auth/signup``` endpoint.
* With Postman, head to the ```POST /api/v1/auth/forgot-password``` endpoint and supply the user's email (ensure the email supplied is functional).
* Check inbox (or spam folder) for the reset password email and click the 'Reset Password' button or extract the reset link.
* Using Postman, make a POST request to the reset link to complete the password reset. ```POST /api/v1/auth/reset-password?key=<resetToken>```
#### What are the relevant pivotal tracker stories?
#164977335
#### Screenshots (if appropriate)
![memoire-reset-password-email](https://user-images.githubusercontent.com/29978455/55543671-67b36e00-56c1-11e9-9f18-9264a49aa815.png)
